lib/rich_text.rb

```diff
@@ -1,5 +1,53 @@
 # frozen_string_literal: true
 
+module Kramdown
+  module Converter
+    class HtmlBidi < Html
+      # Add dir="auto" for certain elements.
+
+      # Elements marked "container" or "child" are not viable in modern browsers unless and until the behaviour of dir="auto" changes:
+      # https://github.com/whatwg/html/issues/10097#issuecomment-2746324449
+      # instead, without dir, they'll inherit their parent's direction (for containers, likely the website's root direction)
+      [
+        "p", "header",
+
+        "codespan", "codeblock", # inner text will usually be LTR and should render accordingly
+
+        # "blockquote", # container
+
+        "ul", "ol", # match Discourse; sometimes container, but this is probably the least bad solution
+        # "li", # child; sometimes container
+
+        "table", # similar to Discourse, where the <div> parent of <table> gets dir="auto" (no parent div in kramdown)
+        # "td", # child
+        # "th", # child; not actually a thing in kramdown but hypothetically might be in the future
+
+        "dl",
+        # "dd", "dt", # child - since it's similar to a list, it has the same behavior as a list
+
+        "math" # don't know how this ends up, but dir="auto" is probably correct
+      ].each do |name|
+        define_method :"convert_#{name}" do |el, indent|
+          attr_bak = el.attr.dup # can't avoid mutating the attr hash, so make a backup
+          el.attr["dir"] = "auto" unless el.attr.key?("dir") # if by some miracle dir is already defined, don't override it
+          ret = super(el, indent)
+          el.attr.replace(attr_bak) # restore backup
```

Comment:
What happens if we don't restore the original attributes? Does the element get reused in a way that breaks things?

I can't say for sure as I couldn't find anything that broke when that restore was removed but would just doing `el = el.dup` as the first thing to duplicate the element be a workable alternative?